From d4ac2edfc6b18c14f8faf7e00649fb2b33e6727d Mon Sep 17 00:00:00 2001
From: Spencer Hill <shill@d3engineering.com>
Date: Fri, 22 Sep 2023 13:31:33 -0400
Subject: [PATCH 05/10] media: i2c: Add imx390 driver

---
 drivers/media/i2c/Kconfig  | 11 +++++++++++
 drivers/media/i2c/Makefile |  1 +
 2 files changed, 12 insertions(+)

diff --git a/drivers/media/i2c/Kconfig b/drivers/media/i2c/Kconfig
index a585c376b72c..605062bb40e6 100644
--- a/drivers/media/i2c/Kconfig
+++ b/drivers/media/i2c/Kconfig
@@ -881,6 +881,17 @@ config VIDEO_IMX355
 	  To compile this driver as a module, choose M here: the
 	  module will be called imx355.
 
+config VIDEO_IMX390
+    tristate "Sony IMX390 sensor support"
+    depends on I2C && VIDEO_V4L2 && VIDEO_V4L2_SUBDEV_API
+    depends on MEDIA_CAMERA_SUPPORT
+    help
+      This is a Video4Linux2 sensor driver for the Sony
+      IMX390 camera.
+
+      To compile this driver as a module, choose M here: the
+      module will be called imx390.
+
 config VIDEO_IMX412
 	tristate "Sony IMX412 sensor support"
 	depends on OF_GPIO
diff --git a/drivers/media/i2c/Makefile b/drivers/media/i2c/Makefile
index 421b9b2ec53d..56f0472ad2e9 100644
--- a/drivers/media/i2c/Makefile
+++ b/drivers/media/i2c/Makefile
@@ -127,6 +127,7 @@ obj-$(CONFIG_VIDEO_IMX319)	+= imx319.o
 obj-$(CONFIG_VIDEO_IMX334)	+= imx334.o
 obj-$(CONFIG_VIDEO_IMX335)	+= imx335.o
 obj-$(CONFIG_VIDEO_IMX355)	+= imx355.o
+obj-$(CONFIG_VIDEO_IMX390)	+= imx390.o
 obj-$(CONFIG_VIDEO_IMX412)	+= imx412.o
 obj-$(CONFIG_VIDEO_ISX031)	+= isx031.o
 obj-$(CONFIG_VIDEO_MAX9286)	+= max9286.o
-- 
2.42.0

