From 0a12c6676be974a512f7848e875ecf20a64ea2ea Mon Sep 17 00:00:00 2001
From: Josh Watts <jwatts@d3engineering.com>
Date: Tue, 7 Nov 2023 16:44:48 -0500
Subject: [PATCH 08/14] imx390: modernize

---
 drivers/media/i2c/imx390.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/imx390.c b/drivers/media/i2c/imx390.c
index bc434ec6e099..d2b55abf1e6a 100644
--- a/drivers/media/i2c/imx390.c
+++ b/drivers/media/i2c/imx390.c
@@ -1913,7 +1913,7 @@ static int imx390_identify_module(struct imx390 *imx390)
 	return 0;
 }
 
-static int imx390_remove(struct i2c_client *client)
+static void imx390_remove(struct i2c_client *client)
 {
 	struct v4l2_subdev *sd = i2c_get_clientdata(client);
 	struct imx390 *imx390 = to_imx390(sd);
@@ -1924,7 +1924,7 @@ static int imx390_remove(struct i2c_client *client)
 	pm_runtime_disable(&client->dev);
 	mutex_destroy(&imx390->mutex);
 
-	return 0;
+	return;
 }
 
 irqreturn_t imx390_threaded_irq_fn(int irq, void *dev_id)
-- 
2.42.0

